src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -701,14 +701,33 @@ public static bool SetComObjectData(object obj, object key, object? data)
         [MethodImpl(MethodImplOptions.InternalCall)]
         public static extern bool IsTypeVisibleFromCom(Type t);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern int /* HRESULT */ QueryInterface(IntPtr /* IUnknown */ pUnk, ref Guid iid, out IntPtr ppv);
+        public static unsafe int QueryInterface(IntPtr pUnk, ref Guid iid, out IntPtr ppv)
+        {
+            if (pUnk == IntPtr.Zero)
+                throw new ArgumentNullException(nameof(pUnk));
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern int /* ULONG */ AddRef(IntPtr /* IUnknown */ pUnk);
+            fixed (Guid* pIID = &iid)
+            fixed (IntPtr* p = &ppv)
+            {
+                return ((delegate * stdcall <IntPtr, Guid*, IntPtr*, int>)(*(*(void***)pUnk + 0 /* IUnknown.QueryInterface slot */)))(pUnk, pIID, p);
```

Comment:
This is soon to be invalid syntax. It might make sense to wait for the update?

/cc @333fred @jkoritzinsky

Reply:
Acknowledged at https://github.com/dotnet/runtime/pull/39752#issuecomment-662181763.